pkg/process/util/log_limit.go

```diff
@@ -0,0 +1,43 @@
+package util
+
+import (
+	"time"
+)
+
+// LogLimit is a utility that can be used to avoid logging noisily
+type LogLimit struct {
+	firstN   int
+	interval time.Duration
+	next     time.Time
+}
+
+// NewLogLimit creates a LogLimit where shouldLog will return
+// true the first N times it is called, and will return true once every
+// interval thereafter.
+func NewLogLimit(n int, interval time.Duration) *LogLimit {
+	return &LogLimit{
+		firstN:   n,
+		interval: interval,
+		next:     time.Now().Add(interval),
+	}
+}
+
+// ShouldLog returns true if the caller should log
+func (l *LogLimit) ShouldLog() bool {
+	return l.shouldLogTime(time.Now())
```

Comment:
I like the idea of this structure though I think it could be improved at the cost of accuracy.

This specifically could potentially create a lot of calls to `time.Now()`. Thoughts on just having an internal timer routine that updates `now` every minute or something and use that?